project/src/components/app/app.tsx

```diff
@@ -1,5 +1,12 @@
-function App(): JSX.Element {
-  return <p>Hello, world!</p>;
+import Main from '../pages/main';
+
+type Props = {
+  amount: number,
+  locations: string[]
+}
+
+function App(props: Props): JSX.Element {
+  return <Main { ...props} />;
```

Comment:
```suggestion
  return <Main  />;
```
так делать не нужно. передавай пропсы явно